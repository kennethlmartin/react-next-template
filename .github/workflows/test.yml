name: Test
on: pull_request

jobs:
  unit:
    name: Unit
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Setup Node.js
        uses: actions/setup-node@v1
        with:
          node-version: 15
      - name: Install dependencies
        run: npm ci
      - name: Run unit tests
        run: npm test
  coverage:
    name: Coverage
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Setup Node.js
        uses: actions/setup-node@v1
        with:
          node-version: 15
      - name: Install dependencies
        run: npm ci
      - name: Run test:coverage
        uses: mattallty/jest-github-action@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
