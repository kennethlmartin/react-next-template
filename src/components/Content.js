import styled from '@emotion/styled';

export const Content = styled.div`
  margin: 0 auto;
  max-width: 1000px;
  padding: 0 5%;
  width: 100%;
`;
