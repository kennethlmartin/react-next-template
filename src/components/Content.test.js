import { render } from 'test/react-testing-library';

import { Content } from './Content';

it('renders without crashing', () => {
  const { container } = render(<Content />);
  expect(container).toMatchSnapshot();
});
