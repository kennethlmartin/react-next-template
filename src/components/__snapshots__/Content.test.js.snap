// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.emotion-0 {
  margin: 0 auto;
  max-width: 1000px;
  padding: 0 5%;
  width: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
